builtin/providers/aws/resource_aws_elastic_beanstalk_environment.go

```diff
@@ -0,0 +1,526 @@
+	conn := meta.(*AWSClient).elasticbeanstalkconn
+
+	if d.HasChange("description") {
+		if err := resourceAwsElasticBeanstalkEnvironmentDescriptionUpdate(conn, d); err != nil {
+			return err
+		}
+	}
+
+	if d.HasChange("solution_stack_name") {
+		if err := resourceAwsElasticBeanstalkEnvironmentSolutionStackUpdate(conn, d); err != nil {
+			return err
+		}
+	}
+
+	if d.HasChange("setting") {
+		if err := resourceAwsElasticBeanstalkEnvironmentOptionSettingsUpdate(conn, d); err != nil {
+			return err
+		}
+	}
+
+	return resourceAwsElasticBeanstalkEnvironmentRead(d, meta)
+}
+
+func resourceAwsElasticBeanstalkEnvironmentDescriptionUpdate(conn *elasticbeanstalk.ElasticBeanstalk, d *schema.ResourceData) error {
+	name := d.Get("name").(string)
+	desc := d.Get("description").(string)
+	envId := d.Id()
+
+	log.Printf("[DEBUG] Elastic Beanstalk application: %s, update description: %s", name, desc)
+
+	_, err := conn.UpdateEnvironment(&elasticbeanstalk.UpdateEnvironmentInput{
+		EnvironmentId: aws.String(envId),
+		Description:   aws.String(desc),
+	})
+
+	return err
+}
+
+func resourceAwsElasticBeanstalkEnvironmentOptionSettingsUpdate(conn *elasticbeanstalk.ElasticBeanstalk, d *schema.ResourceData) error {
+	name := d.Get("name").(string)
+	envId := d.Id()
+
+	log.Printf("[DEBUG] Elastic Beanstalk application: %s, update options", name)
+
+	req := &elasticbeanstalk.UpdateEnvironmentInput{
+		EnvironmentId: aws.String(envId),
+	}
+
+	if d.HasChange("setting") {
+		o, n := d.GetChange("setting")
+		if o == nil {
+			o = &schema.Set{F: optionSettingValueHash}
+		}
+		if n == nil {
+			n = &schema.Set{F: optionSettingValueHash}
+		}
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		req.OptionSettings = extractOptionSettings(ns.Difference(os))
+	}
+
+	if _, err := conn.UpdateEnvironment(req); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsElasticBeanstalkEnvironmentSolutionStackUpdate(conn *elasticbeanstalk.ElasticBeanstalk, d *schema.ResourceData) error {
+	name := d.Get("name").(string)
+	solutionStack := d.Get("solution_stack_name").(string)
+	envId := d.Id()
+
+	log.Printf("[DEBUG] Elastic Beanstalk application: %s, update solution_stack_name: %s", name, solutionStack)
+
+	_, err := conn.UpdateEnvironment(&elasticbeanstalk.UpdateEnvironmentInput{
+		EnvironmentId:     aws.String(envId),
+		SolutionStackName: aws.String(solutionStack),
+	})
+
+	return err
+}
+
+func resourceAwsElasticBeanstalkEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticbeanstalkconn
+
+	app := d.Get("application").(string)
+	envId := d.Id()
+
+	log.Printf("[DEBUG] Elastic Beanstalk environment read %s: id %s", d.Get("name").(string), d.Id())
+
+	resp, err := conn.DescribeEnvironments(&elasticbeanstalk.DescribeEnvironmentsInput{
+		ApplicationName: aws.String(app),
+		EnvironmentIds:  []*string{aws.String(envId)},
+	})
+
+	if err != nil {
```

Comment:
Actually, after digging a bit into this, it seems the EB api returns a `400` because some of the underlying API calls triggered by an environment update return a `503`. It seems reducing parallelism helps, but I'm not too sure if there's anything to be done here